dd-trace-ot/src/main/java/datadog/opentracing/DDSpan.java

```diff
@@ -144,9 +150,12 @@ public void setErrorMeta(final Throwable error) {
 
   private boolean extractError(final Map<String, ?> map) {
     if (map.get(ERROR_OBJECT) instanceof Throwable) {
-      final Throwable error = (Throwable) map.get(ERROR_OBJECT);
-      setErrorMeta(error);
-      return true;
+      // if custom instrumentation don't use setErrorMeta
+      if (map.get(ERROR_KIND) == null && map.get(MESSAGE) == null && map.get(STACK) == null) {
+        final Throwable error = (Throwable) map.get(ERROR_OBJECT);
```

Comment:
Get once to evaluate the `if` and cast as `Throwable` inside the if:

```java
Object obj = map.get(ERROR_OBJECT);
if (obj instanceof Throwable) {
  Throwable error = (Throwable) obj;
  setErrorMeta(error);
  return true;
}
```

Map lookups are expensive, and instrumentation must remain as cheap as possible.